### DISCUSSION TITLE
Async property getters/setters

### DISCUSSION THREAD

**Discussion by ekolis** [dotnet/csharplang#8859] [Language Ideas]
Since property getters and setters are syntactic sugar for methods, I'd like to be able to declare getters and setters as async:

```
public int SomeCachedThing
{
    get => someCachedValue;
    async set => await StoreItInCache(value);
}
```

**Comment by FaustVX** [comment]
**If** this happen, it should be the whole property `async`.
In compound operators (`await SomeCachedThing += 1`) doesn't make sense to `await` a sync accessor.

**Comment by HaloFour** [comment]
Properties really shouldn't be used for fallible I/O operations that would necessitate `async`.

**Reply by colejohnson66** [reply]
Agree. Properties are supposed to be nothing more than fancier fields.

**Reply by jnm2** [reply]
Agreed, and I think I'd rather see `await SetSomeCachedThingAsync(42);` than `await SomeCachedThing = 42;`.

**Reply by jnm2** [reply]
(Not to mention, `await SomeCachedTask = someTask;` has a meaning today)
